pkg/vm/interop.go

```diff
@@ -1,13 +1,50 @@
 package vm
 
 import (
+	"crypto/sha256"
+	"encoding/binary"
 	"errors"
 	"fmt"
+	"sort"
 )
 
 // InteropFunc allows to hook into the VM.
 type InteropFunc func(vm *VM) error
 
+// InteropFuncPrice represents an interop function with a price.
+type InteropFuncPrice struct {
+	Func  InteropFunc
+	Price int
+}
+
+// interopIDFuncPrice adds an ID to the InteropFuncPrice.
+type interopIDFuncPrice struct {
+	ID uint32
+	InteropFuncPrice
+}
+
+// InteropGetterFunc is a function that returns an interop function-price
+// structure by the given interop ID.
+type InteropGetterFunc func(uint32) *InteropFuncPrice
+
+var defaultVMInterops []interopIDFuncPrice
+
+func getDefaultVMInterop(id uint32) *InteropFuncPrice {
+	n := sort.Search(len(defaultVMInterops), func(i int) bool {
```

Comment:
Why have you decided to use sorted slice instead of map?

Reply:
Map is a more complex structure with some more overhead both in terms of memory and operations required to do things we need to do (think about cache locality also), so slice looks way better to me here. Of course we could benchmark it, but I'm not sure it's worth it.

Reply:
Basically, our problem can be solved with a slice, so why bother with a map?